src/common/transformations/src/transformations/resolve_names_collisions.cpp

```diff
@@ -20,63 +35,28 @@ bool ov::pass::ResolveNameCollisions::run_on_model(const std::shared_ptr<ov::Mod
     std::vector<Node*> nodes_with_conflicts;
     std::unordered_map<std::string, std::list<Node*>> visited_nodes;
 
-    for (const auto& node : model->get_ordered_ops()) {
-        // Detect names collisions only for nodes with autogenerated names
-        const auto& friendly_name = node->get_friendly_name();
-        visited_nodes[friendly_name].emplace_back(node.get());
-    }
+    collect_name_collisions_map(model, visited_nodes);
 
-    for (const auto& l_nodes : visited_nodes) {
-        if (l_nodes.second.size() == 1)
+    for (const auto& it : visited_nodes) {
+        const auto& same_named_ops = it.second;
+        if (same_named_ops.size() < 2) {
             continue;
-        const size_t nodes_size = l_nodes.second.size();
-        bool has_public_node = false;  // Parameter, Result ans Sinks
-        size_t i(0);
-        for (auto* node : l_nodes.second) {
-            i++;
-            // Skip the last node if we don't have public nodes with collisions
-            if (i == nodes_size && !has_public_node)
-                break;
-            if (dynamic_cast<const ov::op::v0::Result*>(node)) {
-                // Result is a service node
+        }
+
+        int64_t cnt = 2;
+        for (const auto& op : same_named_ops) {
+            // check if op has OV autogenerated friendly name. unique and friendly names have to be equal.
+            bool is_autogenerated = op->m_friendly_name.empty();
```

Comment:
can we get `it.first` and check that it is empty?

Reply:
no, we can't. ->get_friendly_name() method will return autogenerated name